tests/intensive/plot_tests.py

```diff
@@ -0,0 +1,349 @@
+"""
+Plot tests.
+
+You must run these tests interactively as follows::
+
+    pytest tests/intensive/plot_tests.py -s -k <test_case>
+
+| Copyright 2017-2023, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import random
+import unittest
+
+import numpy as np
+
+import fiftyone as fo
+import fiftyone.brain as fob
+import fiftyone.zoo as foz
+from fiftyone import ViewField as F
+
+
+###############################################################################
+# scatterplot()
+###############################################################################
+
+
+def test_scatterplot():
+    _test_scatterplot("plotly")
+
+
+def test_scatterplot_mpl():
+    _test_scatterplot("matplotlib")
+
+
+def _test_scatterplot(backend):
+    dataset = foz.load_zoo_dataset("quickstart")
+
+    points = np.random.randn(len(dataset), 2)
+    labels = dataset.values("uniqueness")
+    sizes = dataset.values(F("ground_truth.detections").length())
+
+    plot = fo.scatterplot(
+        points=points,
+        labels=labels,
+        sizes=sizes,
+        backend=backend,
+    )
+    plot.show()
+
+    input("Press enter to continue...")
+
+
+def test_scatterplot_interactive():
+    _test_scatterplot_interactive("plotly")
+
+
+def test_scatterplot_interactive_mpl():
+    _test_scatterplot_interactive("matplotlib")
+
+
+def _test_scatterplot_interactive(backend):
+    dataset = foz.load_zoo_dataset("quickstart")
+
+    points = np.random.randn(len(dataset), 2)
+
+    plot = fo.scatterplot(
+        points=points,
+        samples=dataset,
+        labels="uniqueness",
+        sizes=F("ground_truth.detections").length(),
+        backend=backend,
+    )
+    plot.show()
+
+    input("Press enter to continue...")
+
+
+def test_scatterplot_by_ids():
+    _test_scatterplot_by_ids("plotly")
+
+
+def test_scatterplot_by_ids_mpl():
+    _test_scatterplot_by_ids("matplotlib")
+
+
+def _test_scatterplot_by_ids(backend):
+    dataset = foz.load_zoo_dataset("quickstart")
+
+    view = dataset.take(51, seed=51)
+    points = np.random.randn(len(view), 2)
+    ids = view.values("id")
+
+    plot = fo.scatterplot(
+        points=points,
+        samples=dataset,
+        ids=ids,
```

Comment:
This is an example of something that wouldn't previously work. User provided `ids` + `points` corresponding to a subset of the `samples=dataset` argument that they provided, but they provided paths/expressions for `labels` and `sizes` arguments.

Previously this would fail because `dataset.values()` would naively be used (which would result in too many labels/sizes). Now the `ids` argument is used to lookup the correct labels/sizes in the correct order corresponding to points/ids.